### PR TITLE
Add CORS to routes

### LINKED ISSUE #5: Allow Cross-Origin requests
From https://github.com/pypa/warehouse/issues/4687:

> **What's the problem this feature will solve?**
> I'm a developer of [pyodide](http://github.com/iodide-project/pyodide), a port of CPython and the scientific Python stack to WebAssembly.  One feature we'd love to add is the ability to install (pure Python) wheels directly from PyPI into the browser (see https://github.com/iodide-project/pyodide/pull/147).  Unfortunately, this isn't currently possible, since requesting files from `files.pythonhosted.org` is blocked by CORS.  There have been a number of PRs (e.g. #578) that have made CORS work well for the PyPI JSON API, and indeed this is working great for our scenario, so I wonder if you'd be willing to expand that to include the content itself.
>
> **Describe the solution you'd like**
> Adding the appropriate headers to the responses on `files.pythonhosted.org` to allow CORS requests.  Primarily, this would be setting `Access-Control-Allow-Origin` to `*`, though there may be some other required details.  I am *not* an expert in CORS issues.

I don't see any reason why we can't add headers similar to https://github.com/pypa/warehouse/blob/211c43c60aed72db1ace58986cd234d4743fed85/warehouse/legacy/api/json.py#L25-L42 here.

### PR DESCRIPTION
**Pull request by teonbrooks**
closes #5. this adds and uses the aiohttp_cors helper package to enable CORS for the routes.

### CODE REVIEW

**Comment by teonbrooks** [comment]
hi @di, is this what is necessary to correct for the cors issues?

**Comment by di** [comment]
I think all we need is:

```
"Access-Control-Allow-Origin": "*",
"Access-Control-Allow-Methods": "GET",
```

I don't think we necessarily need `Allow-Credentials`, `Expose-Headers` or `Allow-Headers`, although I might be missing something, why did you add them?

@mdboom, perhaps you can chime in here?

**Comment by teonbrooks** [comment]
> I don't think we necessarily need `Allow-Credentials`, `Expose-Headers` or `Allow-Headers`, although I might be missing something, why did you add them?

I think you are correct, just updated the pr

**Comment by mdboom** [comment]
Sorry, I'm not much of a CORS expert, so don't have anything to add.  I'd just suggest merging the ost minimal thing likely to work and we can report back what works or doesn't...

File: requirements.in
Comment (di) [inline requirements.in:3]: This also needs to be compiled into the [`requirements.txt`](https://github.com/pypa/conveyor/blob/master/requirements.txt) file.

File: conveyor/config.py
Comment (di) [inline conveyor/config.py]:
```suggestion
    # Allow cross-origin GETs by default
```

File: conveyor/config.py
Comment (di) [inline conveyor/config.py]:
```suggestion
    # Add the default CORS configuration to all routes
```

**Review by di — CHANGES_REQUESTED** [review]

**Review by di — APPROVED** [review]

File: conveyor/config.py
Comment (ewdurbin) [inline conveyor/config.py]: I think I'd rather see this applied only to the `/packages` routes? We don't want to do this for legacy docs as well do we?

Comment (di) [reply conveyor/config.py]: +1, forgot about the docs routes.

Comment (teonbrooks) [reply conveyor/config.py]: cool. just updated it. i have it working on the subroutes on `/packages`. is that overkill or does applying to the package level push it down downstream?

Comment (ewdurbin) [reply conveyor/config.py]: that looks spot on, I believe each route needs to be CORS'd independently.

**Review by ewdurbin — APPROVED** [review]

**Comment by teonbrooks** [comment]
just curious if there is anything else left on my side that I should address

**Comment by di** [comment]
Hi @teonbrooks, we're just waiting for a review from @dstufft if he's able, otherwise this LGTM and @ewdurbin.

**Comment by mdboom** [comment]
How do I confirm when this has been deployed?  It doesn't look like we're getting CORS headers at these URLs yet.

**Comment by ewdurbin** [comment]
@mdboom whoops! looks like the deploy automation for conveyor broke with the GitHub services deprecations. I'll fix that up now and confirm when this commit is deployed.

**Comment by ewdurbin** [comment]
This is deployed, are you seeing the expected headers for your requests?

**Comment by mdboom** [comment]
Thanks!  This is working great now for the metadata.

Now it looks like we need to address this on the package content itself, which I assume is served through some other mechanism.  For example, the URL:

https://files.pythonhosted.org/packages/d4/6c/8a935e2c7b54a37714656d753e4187ee0631988184ed50c0cf6476858566/snowballstemmer-1.2.1-py2.py3-none-any.whl

has the following headers:

```
HTTP/1.1 200 OK^M
x-amz-id-2: hfgnekIcvgETfsEcwqCcWIHqM8aPhKpVGd+xNIo4LsEyzYucOKQbIrAmtCVXzd3M9Uk5GMarbuQ=^M
x-amz-request-id: 53203498C898622C^M
Last-Modified: Sat, 23 Apr 2016 03:57:53 GMT^M
ETag: "198b60fa6a36d22c6c6f0b9304c6a40b"^M
x-amz-version-id: ZjAIy7CZc1hMcltpkyVR67eyK70o4srA^M
Content-Type: application/octet-stream^M
Server: AmazonS3^M
Cache-Control: max-age=365000000, immutable, public^M
Content-Length: 64341^M
Accept-Ranges: bytes^M
Date: Wed, 24 Apr 2019 19:36:45 GMT^M
Age: 8419414^M
Connection: keep-alive^M
X-Served-By: cache-sea1023-SEA, cache-bwi5039-BWI^M
X-Cache: HIT, HIT^M
X-Cache-Hits: 4, 163^M
X-Timer: S1556134605.230538,VS0,VE0^M
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload^M
X-Frame-Options: deny^M
X-XSS-Protection: 1; mode=block^M
X-Content-Type-Options: nosniff^M
X-Permitted-Cross-Domain-Policies: none^M
X-Robots-Header: noindex^M
```

**Comment by di** [comment]
Seems like maybe there is an outstanding issue, see #8.

**Pull request merged**
